lib/requests/sharing-request.js

```diff
@@ -64,19 +64,26 @@ class ConsentRequest extends AuthRequest {
    * @param res {ServerResponse}
    */
   static async get (req, res) {
-    const request = ConsentRequest.fromParams(req, res)
+    const request = SharingRequest.fromParams(req, res)
 
-    const appOrigin = request.getAppOrigin()
-    // Check if is already registered or is data browser
+    const appUrl = request.getAppUrl()
+    const appOrigin = appUrl.origin
+    const serverUrl = new url.URL(req.app.locals.ldp.serverUri)
+
+    console.log(new url.URL(request.session.subject._id).origin, serverUrl.origin)
+    // Check if is already registered or is data browser or the webId is not on this machine
     if (request.isUserLoggedIn()) {
       if (
-        appOrigin === req.app.locals.ldp.serverUri ||
+        new url.URL(request.session.subject._id).origin !== serverUrl.origin ||
+        (appUrl && request.isSubdomain(serverUrl.host, appUrl.host) && appUrl.protocol === serverUrl.protocol) ||
         await request.isAppRegistered(req.app.locals.ldp, appOrigin, request.session.subject._id)
       ) {
-        request.setUserConsent(appOrigin)
-        request.redirectPostConsent()
+        console.log('We be here')
```

Comment:
This got a bit chatty when trying it out. We shouldn't have `console.log` in at this point, but it could be a `debug` there, but then something more informative.